### PR TITLE
Skip unnecessary CI tests in Baremetal Platform repos

### PR DESCRIPTION
**Pull request by zaneb**
Don't run tests on PRs where only irrelevant files (primarily documentation and `OWNERS`) have changed.

### CODE REVIEW

File: ci-operator/jobs/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master-presubmits.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ presubmits:
       pj-rehearse.openshift.io/can-be-rehearsed: "true"
     name: pull-ci-openshift-metal3-dev-scripts-master-e2e-metal-ipi
     rerun_command: /test e2e-metal-ipi
+    skip_if_only_changed: (^(docs|.github)/)||(\.md$)|((^|/)OWNERS$)
```
Comment (stbenjam) [inline ci-operator/jobs/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master-presubmits.yaml:17]: Does skip_if_only_change require always_run false?

Comment (elfosardo) [reply ci-operator/jobs/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master-presubmits.yaml:17]: yes

**Comment by ardaguclu** [comment]
lgtm

**Pull request merged**